src/ui/src/builder/panels/BuilderCodePanel.vue

```diff
@@ -268,8 +268,20 @@ async function handleSave() {
 
 	isDisabled.value = true;
 
+	let pathSegments: string[] | undefined = undefined;
+	if (isRenaming.value) {
+		pathSegments = filename.value.split("/");
```

Comment:
I guess you can improve the readability by avoiding the re-assignation 

```suggestion
	const pathSegments = isRenaming.value ? filename.value.split("/") : undefined;
	if (pathSegments) {
```